src/main/java/com/softserve/academy/spaced/repetition/service/impl/AccountServiceImpl.java

```diff
@@ -36,43 +30,24 @@
     @Autowired
     private UserService userService;
     @Autowired
-    private NumberOfPostponedDaysValidator numberOfPostponedDaysValidator;
-    @Autowired
     private MailService mailService;
     @Autowired
     private PasswordEncoder passwordEncoder;
-    @Autowired
-    private MessageSource messageSource;
-    private final Locale locale = LocaleContextHolder.getLocale();
-
-    @Override
-    public void updateAccount(Account account) {
-        accountRepository.save(account);
-    }
 
     @Override
     @Transactional
-    public LearningRegime getLearningRegime() throws NotAuthorisedUserException {
-        return userService.getAuthorizedUser().getAccount().getLearningRegime();
+    public Account updateAccountDetails(Account acc) throws NotAuthorisedUserException {
+        Account account = userService.getAuthorizedUser().getAccount();
+        account.setRememberingLevels(acc.getRememberingLevels());
+        account.setCardsNumber(acc.getCardsNumber());
+        account.setLearningRegime(acc.getLearningRegime());
+        return accountRepository.save(account);
     }
 
     @Override
     @Transactional
-    public void updateLearningRegime(String learningRegime) throws NotAuthorisedUserException,
-            IllegalArgumentException {
-        boolean learningRegimeFound = Arrays.stream(LearningRegime.values())
-                .anyMatch(LearningRegime.valueOf(learningRegime)::equals);
-
-        if(!learningRegimeFound) {
-            throw new IllegalArgumentException(messageSource.getMessage("message.exception.learningRegimeNotValid",
-                    new Object[]{learningRegime}, locale));
-        }
-
-        LearningRegime regime = LearningRegime.valueOf(learningRegime);
-
-        Account account = accountRepository.findOne(userService.getAuthorizedUser().getAccount().getId());
-        account.setLearningRegime(regime);
-        accountRepository.save(account);
+    public Account getAccountDetails() throws NotAuthorisedUserException {
+        return userService.getAuthorizedUser().getAccount();
```

Comment:
Maybe will be better to create some variable of `User` type and after that get `Account` object from it.
Does `@Transactional` annotation is necessary here?